### ISSUE TITLE
EU GDPR - General Data Protection Regulation

### ISSUE THREAD

**Issue by syntaxerror** [arastta/arastta#757]
Enforcement date for EU GDPR is 25 May 2018: https://www.eugdpr.org/

Has anyone looked at how this will affect Arastta, and stores running on it?

As far as I have seen there is a need to clarify which personal data who is being collected and stored, where IP also is considered to be personal data, for both random visitors and registered customers, plus guest checkouts. There might also be need for some automatic deleting of stored IP-addresses after a given time.

The registered customers should also have access to their stored data (including IP addresses), or being able to request this info - meaning store owner should have access to get it all out in a simple report.

Luckily the customers already have access to most of their data through their account (excluding IP's), and can change it from there too. But there should probably also be a easy way for them to delete their account, and all data except the data on the orders - needed for the order delivery and accounting (book keeping rules being stronger than GDPR in that matter).

Thoughts?

_Edit/Addition: Some of it is covered by this OC extension, if it works on Arastta I don't know. But it serves as a source of ideas for some of the parts needed to comply to EU GDPR: https://www.opencart.com/index.php?route=marketplace/extension/info&extension_id=32993_

**Comment by denisdulici** [comment]
Such additions should be done through extensions and it seems the one you shared solves the problem, thanks Rune.

**Comment by syntaxerror** [comment]
I strongly disagree, as this is a requirement for all running a store who is selling to EU citizens.

I encourage you to reopen and discuss this more thoroughly, let others also join in with their opinion.

**Comment by denisdulici** [comment]
Reopened, feel free to discuss.

**Comment by syntaxerror** [comment]
Auto cleaning old temp orders would probably be required, as storing of unneeded data seems illegal.
https://arastta.org/ideas/idea/44-auto-clean-old-temp-orders-abandoned-carts

**Comment by syntaxerror** [comment]
A nice way to deal with it, adding value to the project, by releasing an official paid for extension:
https://addons.prestashop.com/en/legal/32323-official-gdpr-compliance-by-prestashop-16.html
